_includes/_location-info.html

```diff
@@ -2,24 +2,52 @@
-      <span class="font-family-base-bold font-size-smaller text-uppercase">Service Times</span>
       <p class="flush font-size-small">
-      {% if site.data.service_times.size != 0 %}
-        {% assign service_times_str = site.data.service_times[page.name] %}
-        {% assign service_times_array = service_times_str | split: '|' %}
-        {% for time in service_times_array %}
-          {% assign parts = time | split: '^' %}
-          {% if parts[0] == 'Anywhere' %}
-            <div>{{ parts[0] }} {{ parts[1] | replace: '%2C', ',' }}{% unless forloop.last %}<br>{% endunless %}</div>
-          {% else %}
-            <div>{{ parts[0] }}: {{ parts[1] | replace: '%2C', ',' }}{% unless forloop.last %}<br>{% endunless %}</div>
-          {% endif %}
+        {% assign special_service_times = "" | split: "" %}
+        {% assign standard_service_times = "" | split: "" %}
+        {% assign special_service_time_label = "" %}
+        {% assign standard_service_time_label = "" %}
+        {% assign current_date = "now" | date: "%s" %}
+    
+        {% if page.service_times %}
+          {% for time in page.service_times %}
+            {% if time.is_special_service_time == true %}
+              {% assign start_date = time.special_service_start_date | date: "%s" %}
+              {% assign end_date = time.special_service_end_date | date: "%s" %}
+              {% if current_date >= start_date and current_date <= end_date %}
+                {% assign special_service_time_label = time.service_time_label %}
+                {% assign special_service_times = special_service_times | concat: time.service_times %}
+              {% endif %}
+            {% else %}
+              {% assign standard_service_time_label = time.service_time_label %}
+              {% assign standard_service_times = standard_service_times | concat: time.service_times %}
+            {% endif %}
+          {% endfor %}
+        {% endif %}
+    
+        {% if standard_service_time_label %}
+          <span class="font-family-base-bold">{{ standard_service_time_label }}</span><br>
+        {% endif %}        
+        {% for service_time in standard_service_times %}
+          {{ service_time }}
+          <!-- don't render the <br> if it's the last line rendered -->
+          {% if forloop.index != standard_service_times.size %}<br>{% endif %}
+        {% endfor %}
+    
+        {% if current_date >= start_date and current_date <= end_date %}
+        <br><br>
+        {% endif %}
+    
+        {% if special_service_time_label %}
+          <span class="font-family-base-bold">{{ special_service_time_label }}</span><br>
+        {% endif %}
+        {% for service_time in special_service_times %}
+          {{ service_time }}
+          <!-- don't render the <br> if it's the last line rendered -->
+          {% if forloop.index != special_service_times.size %}<br>{% endif %}
         {% endfor %}
```

Comment:
one issue I see here with this solution is that there was a desire to have the order of the service times display change based on the order service times were added on the location.